File: 코딩 기초 트레이닝/빈 배열에 추가, 삭제하기-p181860/빈 배열에 추가, 삭제하기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181860

def solution(arr, flag):
    answer = []

    for index, f in enumerate(flag):
        if(f):
            a = [ arr[index] for a in range(arr[index] * 2)]
            answer += a
        else:
            answer = answer[0: len(answer) - arr[index]]

    return answer

File: 코딩 기초 트레이닝/빈 배열에 추가, 삭제하기-p181860/빈 배열에 추가, 삭제하기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181860

function solution(arr, flag) {
    let X = [];

    flag.forEach((f, index) => {
        if(f) {
            const a = new Array(arr[index] * 2).fill(arr[index]);

            X = X.concat(...a);
        } else {
            X = X.slice(0, X.length-arr[index]);
        }
    });

    return X;
}
